src/utils/privateRoute.js

```diff
@@ -1,15 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Route, Redirect } from 'react-router-dom';
+import axiosWithAuth from './axiosWithAuth';
+
+/* eslint-disable no-undef */
+const BACKEND_URL =
+  process.env.REACT_APP_DEPLOYED_URL || 'http://localhost:5000';
 
 export const PrivateRoute = ({ component: Component, ...rest }) => {
+  const [activeSession, setActiveSession] = useState(false);
+
+  // ask the db if this user has an active session
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`${BACKEND_URL}/api/user`)
+      .then((response) => {
+        if (response.data.user) {
+          setActiveSession(true);
+        }
+      })
+      .catch((err) => {
+        console.log('privateRoute err', err);
+      });
+  }, []);
+
   return (
     <Route
       {...rest}
       render={(props) => {
-        if (localStorage.getItem('token')) {
+        if (activeSession) {
           return <Component {...props} />;
         } else {
-          return <Redirect to="/welcome" />;
+          <Redirect to="/welcome" />;
         }
```

Comment:
Awesome bug tackle!

Reply:
i just pushed a new commit that changed this file. is solved an issue where state was being changed on an unmounted component.